pkg/windows/calico.go

```diff
@@ -52,147 +137,235 @@ const (
+	c.DataDir = dataDir
+
+	if err := c.initializeConfig(ctx, nodeConfig, restConfig); err != nil {
+		return err
+	}
+
+	if err := c.overrideCalicoConfigByHelm(restConfig); err != nil {
+		return err
+	}
+
+	if err := c.writeConfigFiles(nodeConfig.AgentConfig.CNIConfDir, nodeConfig.AgentConfig.NodeName); err != nil {
+		return err
 	}
 
-	client, err := coreClient(restConfig)
+	logrus.Info("Generating HNS networks, please wait")
+	return c.generateCalicoNetworks()
+}
+
+// initializeConfig sets the default configuration in CNIConfig
+func (c *Calico) initializeConfig(ctx context.Context, nodeConfig *daemonconfig.Node, restConfig *rest.Config) error {
+	c.CNICfg = &CalicoConfig{}
+	platformType, err := getPlatformType()
 	if err != nil {
-		return nil, err
+		return err
 	}
-	serviceAccounts := client.CoreV1().ServiceAccounts(CalicoSystemNamespace)
 
-	req := authv1.TokenRequest{
-		Spec: authv1.TokenRequestSpec{
-			Audiences:         []string{version.Program},
-			ExpirationSeconds: pointer.Int64(60 * 60 * 24 * 365),
+	c.CNICfg = &CalicoConfig{
+		Name:                  "Calico",
+		OverlayNetName:        "Calico",
+		Hostname:              nodeConfig.AgentConfig.NodeName,
+		NodeNameFile:          filepath.Join("c:\\", c.DataDir, "agent", CalicoNodeNameFileName),
+		KubeNetwork:           "Calico.*",
+		Mode:                  "vxlan",
+		ServiceCIDR:           nodeConfig.AgentConfig.ServiceCIDR.String(),
+		DNSServers:            nodeConfig.AgentConfig.ClusterDNS.String(),
+		DNSSearch:             "svc." + nodeConfig.AgentConfig.ClusterDomain,
+		DatastoreType:         "kubernetes",
+		Platform:              platformType,
+		StartUpValidIPTimeout: 90,
+		IP:                    nodeConfig.AgentConfig.NodeIP,
+		IPAutoDetectionMethod: "first-found",
+		Felix: FelixConfig{
+			Metadataaddr:    "none",
+			Vxlanvni:        "4096",
+			MacPrefix:       "0E-2A",
+			LogSeverityFile: "none",
+			LogSeveritySys:  "none",
+		},
+		CNI: CalicoCNIConfig{
+			BinDir:   nodeConfig.AgentConfig.CNIBinDir,
+			ConfDir:  nodeConfig.AgentConfig.CNIConfDir,
+			IpamType: "calico-ipam",
+			Version:  "0.3.1",
 		},
 	}
 
-	token, err := serviceAccounts.CreateToken(ctx, calicoNode, &req, metav1.CreateOptions{})
+	c.CNICfg.KubeConfig, err = c.createCalicoKubeConfig(ctx, restConfig)
 	if err != nil {
-		return nil, errors.Wrapf(err, "failed to create token for service account (%s/%s)", CalicoSystemNamespace, calicoNode)
+		return err
 	}
-	calicoKubeConfig.Token = token.Status.Token
 
-	cfg := &CNIConfig{NodeConfig: nodeConfig, NetworkName: "Calico", BindAddress: nodeConfig.AgentConfig.NodeIP}
+	return nil
+}
 
-	hc, err := helm.NewFactoryFromConfig(restConfig)
+// writeConfigFiles writes the three required files by Calico
+func (c *Calico) writeConfigFiles(CNIConfDir string, NodeName string) error {
+
+	// Create CalicoKubeConfig and CIPAutoDetectionMethodalicoConfig files
+	err := c.renderCalicoConfig(c.CNICfg.KubeConfig.Path, calicoKubeConfigTemplate)
```

Comment:
Single err returns should remain scope locked on a single line.

Reply:
Correct again, I missed this